backend/pkg/kafka/consumer.go

```diff
@@ -80,12 +85,22 @@ func (s *Service) FetchMessages(ctx context.Context, progress IListMessagesProgr
 	}
 	defer client.Close()
 
-	// 2. Setup JavaScript interpreter
-	isMessageOK, err := s.setupInterpreter(consumeRequest.FilterInterpreterCode)
-	if err != nil {
-		s.Logger.Error("failed to setup interpreter", zap.Error(err))
-		progress.OnError(fmt.Sprintf("failed to setup interpreter: %v", err.Error()))
-		return err
+	// 2. Create consumer workers
+	jobs := make(chan *kgo.Record, 100)
+	resultsCh := make(chan *TopicMessage, 100)
+	defer close(jobs)
+	workerCtx, cancel := context.WithCancel(ctx)
+
+	for i := 0; i < 4; i++ {
+		// Setup JavaScript interpreter
+		isMessageOK, err := s.setupInterpreter(consumeRequest.FilterInterpreterCode)
```

Comment:
I noticed that `setupInterpreter` compiles the Javascript and then runs it for every record, yet the compilation can be saved and instead the code can be switched to `vm.RunProgram(<saved compiled script>)`. That may be worth pursuing to cut down on some latency / cpu.